database/insertData/_helpers/core_mutations.js

```diff
@@ -149,15 +149,32 @@ exports.coreActions = `
         operator: "AND"
         children: [
           {
-            operator: "="
+            operator: "OR",
             children: [
               {
-                operator: "objectProperties"
+                operator: "=",
                 children: [
-                  "applicationData.reviewData.latestDecision.decision"
+                  {
+                    operator: "objectProperties",
+                    children: [
+                      "applicationData.reviewData.latestDecision.decision"
+                    ]
+                  },
+                  "CONFORM"
+                ]
+              },
+              {
+                operator: "=",
+                children: [
+                  {
+                    operator: "objectProperties",
+                    children: [
+                      "applicationData.reviewData.latestDecision.decision"
+                    ]
+                  },
+                  "NON_CONFORM"
```

Comment:
As we discussed, the stage needs to NOT increment if the decision is "NON_CONFORM" and the next stage is not Final Decision. This logic should probably be inside the incrementStage action, so maybe make a new issue for that, or perhaps we need to discuss it to agree on a definitive solution to handling this logic.

Reply:
I think we can deal with this on issue #447